core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java

```diff
@@ -64,11 +65,15 @@ public void onHalfClose() {
             }
           } catch (UnauthenticatedException e) {
             call.close(Status.UNAUTHENTICATED, headers);
+            callClosed = true;
           }
         } else {
           call.close(Status.UNAUTHENTICATED, headers);
+          callClosed = true;
+        }
+        if (!callClosed) {
+          super.onHalfClose();
```

Comment:
could add a comment as to why `super.onHalfClose()` has to be called, in these circumstances?